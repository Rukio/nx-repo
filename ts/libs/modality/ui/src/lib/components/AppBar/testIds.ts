export const APP_BAR_TEST_IDS = {
  STATION_LINK: 'app-bar-station-link',
};
