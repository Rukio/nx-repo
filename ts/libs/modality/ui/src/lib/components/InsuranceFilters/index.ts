export { default as InsuranceFilters } from './InsuranceFilters';
export { INSURANCE_FILTERS_TESTS_IDS } from './testIds';
