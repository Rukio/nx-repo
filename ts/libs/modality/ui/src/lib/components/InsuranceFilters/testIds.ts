const insuranceFiltersSelectOptionPrefixText =
  'insurance-filters-filter-by-option';

export const INSURANCE_FILTERS_TESTS_IDS = {
  SEARCH_FIELD: 'insurance-filters-search-field',
  FILTER_BY_SELECT: 'insurance-filters-filter-by-select',
  FILTER_BY_OPTION_PREFIX: 'insurance-filters-filter-by-option',
  getFilterByOptionTestId: (insuranceClassificationId: number) =>
    `${insuranceFiltersSelectOptionPrefixText}-${insuranceClassificationId}`,
};
