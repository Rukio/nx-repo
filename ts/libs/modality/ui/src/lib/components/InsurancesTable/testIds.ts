const tableCellModalityPrefix = 'insurances-table-cell-modality';
const tableInsurancePlansRowPrefixText = 'insurances-table-row';

export const INSURANCE_TABLE_TEST_IDS = {
  TABLE_NAME_SORT_LABEL: 'insurances-table-name-sort-label',
  TABLE_LAST_UPDATED_SORT_LABEL: 'insurances-table-last-updated-sort-label',
  TABLE_HEADER_CELL_NAME: 'insurances-table-header-cell-name',
  TABLE_HEADER_CELL_LAST_UPDATED: 'insurances-table-header-cell-last-updated',
  TABLE_CELL_NAME: 'insurances-table-cell-name',
  TABLE_CELL_IN_PERSON: 'insurances-table-cell-in-person',
  TABLE_CELL_TELE_P: 'insurances-table-cell-tele-p',
  TABLE_CELL_VIRTUAL: 'insurances-table-cell-virtual',
  TABLE_CELL_LAST_UPDATED: 'insurances-table-cell-last-updated',
  TABLE_CELL_HEAD_MODALITY: 'insurances-table-cell-head-modality',
  TABLE_CELL_MODALITY_PREFIX: tableCellModalityPrefix,
  TABLE_PAGINATION: 'insurances-table-pagination',
  getModalityCellTestId: (modalityId: number) =>
    `${tableCellModalityPrefix}-${modalityId}`,
  TABLE_ROOT: 'insurance-plans-table',
  TABLE_ROW_PREFIX: tableInsurancePlansRowPrefixText,
  getInsurancePlanRowTestId: (insurancePlanId: number) =>
    `${tableInsurancePlansRowPrefixText}-${insurancePlanId}`,
  TABLE_NO_RESULTS_TEXT: 'insurances-table-no-results-text',
};
