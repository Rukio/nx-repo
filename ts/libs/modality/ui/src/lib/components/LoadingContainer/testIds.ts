export const LOADING_CONTAINER_TEST_IDS = {
  getProgressTestId: (prefix: string) => `${prefix}-loading-container`,
};
