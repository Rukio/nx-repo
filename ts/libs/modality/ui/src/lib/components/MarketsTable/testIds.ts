const tableMarketRowPrefixText = 'markets-table-row';
const tableCellModalityPrefix = 'markets-table-cell-modality';

export const MARKETS_TABLE_TEST_IDS = {
  TABLE_CELL_NAME: 'markets-table-cell-name',
  TABLE_CELL_HEAD_MODALITY: 'markets-table-cell-head-modality',
  TABLE_CELL_IN_PERSON: 'markets-table-cell-in-person',
  TABLE_CELL_TELE_P: 'markets-table-cell-tele-p',
  TABLE_CELL_VIRTUAL: 'markets-table-cell-virtual',
  TABLE_PAGINATION: 'markets-table-pagination',
  TABLE_ROW_PREFIX: tableMarketRowPrefixText,
  TABLE_CELL_MODALITY_PREFIX: tableCellModalityPrefix,
  TABLE_ROOT: 'markets-table',
  getMarketRowTestId: (marketId: number) =>
    `${tableMarketRowPrefixText}-${marketId}`,
  getModalityCellTestId: (modalityId: number) =>
    `${tableCellModalityPrefix}-${modalityId}`,
};
