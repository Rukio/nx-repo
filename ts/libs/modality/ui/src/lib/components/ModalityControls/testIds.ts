export const MODALITY_CONTROLS_TEST_IDS = {
  CANCEL_BUTTON: 'modality-controls-cancel-button',
  SUBMIT_BUTTON: 'modality-controls-submit-button',
};
