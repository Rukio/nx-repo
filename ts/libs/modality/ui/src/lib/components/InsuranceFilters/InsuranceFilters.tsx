import { useState, MouseEvent, ChangeEventHandler } from 'react';
import {
  TextField,
  Grid,
  Typography,
  SearchIcon,
  Menu,
  MenuItem,
  Chip,
  makeSxStyles,
} from '@*company-data-covered*/design-system';
import { INSURANCE_FILTERS_TESTS_IDS } from './testIds';

type FilterOption = { id: number; name: string };

type Props = {
  filterOptions: FilterOption[];
  filterBy?: FilterOption;
  onFilterByChange: (value?: FilterOption) => void;
  searchText: string;
  onChangeSearch: ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
  disableFilter?: boolean;
};

const makeStyles = () =>
  makeSxStyles({
    filtersRoot: {
      mt: 3,
    },
    filterByRoot: {
      display: 'flex',
      alignItems: 'center',
    },
    filterByLabel: {
      color: 'text.secondary',
      mr: 2,
    },
  });

const InsuranceFilters = ({
  filterOptions = [],
  filterBy,
  onFilterByChange,
  searchText,
  onChangeSearch,
  disableFilter,
}: Props) => {
  const classes = makeStyles();
  const [menuAnchorEl, setMenuAnchorEl] = useState<null | HTMLElement>(null);

  const handleOpenMenu = (event: MouseEvent<HTMLElement>) =>
    setMenuAnchorEl(event.currentTarget);

  const handleCloseMenu = () => setMenuAnchorEl(null);

  const onChangeFilter = (value: FilterOption) => () => {
    const isActiveFilterClicked = filterBy?.id === value.id;
    onFilterByChange(isActiveFilterClicked ? undefined : value);
    handleCloseMenu();
  };

  return (
    <Grid container sx={classes.filtersRoot} spacing={3}>
      <Grid item xs={6.5}>
        <TextField
          InputProps={{ startAdornment: <SearchIcon /> }}
          placeholder="Search by insurance plan name or Athena package ID"
          fullWidth
          defaultValue={searchText}
          onChange={onChangeSearch}
          inputProps={{
            'data-testid': INSURANCE_FILTERS_TESTS_IDS.SEARCH_FIELD,
          }}
        />
      </Grid>
      <Grid item sx={classes.filterByRoot}>
        <Typography variant="body2" sx={classes.filterByLabel}>
          Filter by
        </Typography>
        <Chip
          label={filterBy?.name ?? 'Select Filter'}
          variant="outlined"
          clickable
          onClick={handleOpenMenu}
          data-testid={INSURANCE_FILTERS_TESTS_IDS.FILTER_BY_SELECT}
          disabled={disableFilter}
        />
        <Menu
          open={Boolean(menuAnchorEl)}
          anchorEl={menuAnchorEl}
          onClose={handleCloseMenu}
        >
          {filterOptions.map((option) => (
            <MenuItem
              onClick={onChangeFilter(option)}
              key={`filter-by-${option.id}`}
              data-testid={INSURANCE_FILTERS_TESTS_IDS.getFilterByOptionTestId(
                option.id
              )}
              selected={option.id === filterBy?.id}
            >
              {option.name}
            </MenuItem>
          ))}
        </Menu>
      </Grid>
    </Grid>
  );
};

export default InsuranceFilters;
