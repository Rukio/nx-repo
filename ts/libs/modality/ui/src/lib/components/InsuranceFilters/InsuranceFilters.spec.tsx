import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import InsuranceFilters from './InsuranceFilters';
import { INSURANCE_FILTERS_TESTS_IDS } from './testIds';

const props = {
  filterOptions: [
    { name: 'Classification', id: 1 },
    { name: 'Type', id: 2 },
  ],
  filterBy: { name: 'Classification', id: 1 },
  onFilterByChange: jest.fn(),
  searchText: '',
  onChangeSearch: jest.fn(),
};

const setup = () => {
  return {
    ...render(<InsuranceFilters {...props} />),
    user: userEvent.setup(),
  };
};

describe('<InsuranceFilters />', () => {
  it('should render properly', () => {
    const { asFragment } = setup();

    expect(asFragment()).toMatchSnapshot();
  });

  it('should change search', async () => {
    const { user } = setup();

    const searchField = screen.getByTestId(
      INSURANCE_FILTERS_TESTS_IDS.SEARCH_FIELD
    ) as HTMLInputElement;

    await user.type(searchField, 'search');

    expect(props.onChangeSearch).toHaveBeenCalled();
  });

  it('should change filter option', async () => {
    const { user } = setup();

    const filterByButton = screen.getByTestId(
      INSURANCE_FILTERS_TESTS_IDS.FILTER_BY_SELECT
    );

    await user.click(filterByButton);

    const filterByOptions = await screen.findAllByTestId(
      new RegExp(INSURANCE_FILTERS_TESTS_IDS.FILTER_BY_OPTION_PREFIX)
    );
    expect(filterByOptions.length).toBeTruthy();
    await user.click(filterByOptions[1]);

    expect(props.onFilterByChange).toHaveBeenCalledWith(props.filterOptions[1]);
  });

  it('should deselect active filter option', async () => {
    const { user } = setup();

    const filterByButton = screen.getByTestId(
      INSURANCE_FILTERS_TESTS_IDS.FILTER_BY_SELECT
    );

    await user.click(filterByButton);

    const filterByOptions = await screen.findAllByTestId(
      new RegExp(INSURANCE_FILTERS_TESTS_IDS.FILTER_BY_OPTION_PREFIX)
    );
    expect(filterByOptions.length).toBeTruthy();
    await user.click(filterByOptions[0]);

    expect(props.onFilterByChange).toHaveBeenCalledWith(undefined);
  });
});
