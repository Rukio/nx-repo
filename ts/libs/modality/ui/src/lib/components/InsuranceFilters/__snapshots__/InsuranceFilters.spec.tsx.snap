// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InsuranceFilters /> should render properly 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-top: -24px;
  width: calc(100% + 24px);
  margin-left: -24px;
  margin-top: 24px;
}

.emotion-0>.MuiGrid-item {
  padding-top: 24px;
}

.emotion-0>.MuiGrid-item {
  padding-left: 24px;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 54.166667%;
  -ms-flex-preferred-size: 54.166667%;
  flex-basis: 54.166667%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 54.166667%;
}

@media (min-width:600px) {
  .emotion-1 {
    -webkit-flex-basis: 54.166667%;
    -ms-flex-preferred-size: 54.166667%;
    flex-basis: 54.166667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 54.166667%;
  }
}

@media (min-width:900px) {
  .emotion-1 {
    -webkit-flex-basis: 54.166667%;
    -ms-flex-preferred-size: 54.166667%;
    flex-basis: 54.166667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 54.166667%;
  }
}

@media (min-width:1200px) {
  .emotion-1 {
    -webkit-flex-basis: 54.166667%;
    -ms-flex-preferred-size: 54.166667%;
    flex-basis: 54.166667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 54.166667%;
  }
}

@media (min-width:1536px) {
  .emotion-1 {
    -webkit-flex-basis: 54.166667%;
    -ms-flex-preferred-size: 54.166667%;
    flex-basis: 54.166667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 54.166667%;
  }
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  width: 100%;
}

.emotion-3 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  position: relative;
  border-radius: 4px;
  padding-left: 14px;
}

.emotion-3.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
  cursor: default;
}

.emotion-3:hover .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.87);
}

@media (hover: none) {
  .emotion-3:hover .MuiOutlinedInput-notchedOutline {
    border-color: rgba(0, 0, 0, 0.23);
  }
}

.emotion-3.Mui-focused .MuiOutlinedInput-notchedOutline {
  border-color: #1976d2;
  border-width: 2px;
}

.emotion-3.Mui-error .MuiOutlinedInput-notchedOutline {
  border-color: #d32f2f;
}

.emotion-3.Mui-disabled .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.26);
}

.emotion-4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-5 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
  padding: 16.5px 14px;
  padding-left: 0;
}

.emotion-5::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-5::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-5:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-5::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-5:focus {
  outline: 0;
}

.emotion-5:invalid {
  box-shadow: none;
}

.emotion-5::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-5.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0, 0, 0, 0.38);
}

.emotion-5:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-5:-webkit-autofill {
  border-radius: inherit;
}

.emotion-6 {
  text-align: left;
  position: absolute;
  bottom: 0;
  right: 0;
  top: -5px;
  left: 0;
  margin: 0;
  padding: 0 8px;
  pointer-events: none;
  border-radius: inherit;
  border-style: solid;
  border-width: 1px;
  overflow: hidden;
  min-width: 0%;
  border-color: rgba(0, 0, 0, 0.23);
}

.emotion-7 {
  float: unset;
  width: auto;
  overflow: hidden;
  padding: 0;
  line-height: 11px;
  -webkit-transition: width 150ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: width 150ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-9 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  color: rgba(0, 0, 0, 0.6);
  margin-right: 16px;
}

.emotion-10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  max-width: 100%;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 0.8125rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 32px;
  color: rgba(0, 0, 0, 0.87);
  background-color: rgba(0, 0, 0, 0.08);
  border-radius: 16px;
  white-space: nowrap;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  cursor: default;
  outline: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  padding: 0;
  vertical-align: middle;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: transparent;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid #bdbdbd;
}

.emotion-10::-moz-focus-inner {
  border-style: none;
}

.emotion-10.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-10 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-10.Mui-disabled {
  opacity: 0.38;
  pointer-events: none;
}

.emotion-10 .MuiChip-avatar {
  margin-left: 5px;
  margin-right: -6px;
  width: 24px;
  height: 24px;
  color: #616161;
  font-size: 0.75rem;
}

.emotion-10 .MuiChip-avatarColorPrimary {
  color: #fff;
  background-color: #1565c0;
}

.emotion-10 .MuiChip-avatarColorSecondary {
  color: #fff;
  background-color: #7b1fa2;
}

.emotion-10 .MuiChip-avatarSmall {
  margin-left: 4px;
  margin-right: -4px;
  width: 18px;
  height: 18px;
  font-size: 0.625rem;
}

.emotion-10 .MuiChip-icon {
  margin-left: 5px;
  margin-right: -6px;
  color: #616161;
}

.emotion-10 .MuiChip-deleteIcon {
  -webkit-tap-highlight-color: transparent;
  color: rgba(0, 0, 0, 0.26);
  font-size: 22px;
  cursor: pointer;
  margin: 0 5px 0 -6px;
}

.emotion-10 .MuiChip-deleteIcon:hover {
  color: rgba(0, 0, 0, 0.4);
}

.emotion-10:hover {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-10.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.2);
}

.emotion-10:active {
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
}

.emotion-10.MuiChip-clickable:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

.emotion-10.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-10 .MuiChip-avatar {
  margin-left: 4px;
}

.emotion-10 .MuiChip-avatarSmall {
  margin-left: 2px;
}

.emotion-10 .MuiChip-icon {
  margin-left: 4px;
}

.emotion-10 .MuiChip-iconSmall {
  margin-left: 2px;
}

.emotion-10 .MuiChip-deleteIcon {
  margin-right: 5px;
}

.emotion-10 .MuiChip-deleteIconSmall {
  margin-right: 3px;
}

.emotion-11 {
  overflow: hidden;
  text-overflow: ellipsis;
  padding-left: 12px;
  padding-right: 12px;
  white-space: nowrap;
}

.emotion-12 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 emotion-0"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6.5 emotion-1"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root emotion-2"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart emotion-3"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-4"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart emotion-5"
            data-testid="insurance-filters-search-field"
            id=":r0:"
            placeholder="Search by insurance plan name or Athena package ID"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline emotion-6"
          >
            <legend
              class="emotion-7"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item emotion-8"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 emotion-9"
      >
        Filter by
      </p>
      <div
        class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault emotion-10"
        data-testid="insurance-filters-filter-by-select"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiChip-label MuiChip-labelMedium emotion-11"
        >
          Classification
        </span>
        <span
          class="MuiTouchRipple-root emotion-12"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
